src/scaladoc/scala/tools/nsc/doc/html/Doclet.scala

```diff
@@ -12,9 +12,17 @@ import doclet._
 /** The default doclet used by the scaladoc command line tool
   * when no user-provided doclet is provided. */
 class Doclet extends Generator with Universer {
+  import scala.reflect.internal.Reporter
 
-  def generateImpl() {
-    new html.HtmlFactory(universe, new ScalaDocReporter(universe.settings)).generate()
```

Comment:
sounds right.

could we make the reporter a constructor argument? i see that Doclet is created reflectively. but maybe we can change that code to do something like

```
docletClass. getConstructors()
// find the one that takes a reporter, invoke it
// if it doesn't exist, invoke the no-args one
```